tests/query/test_query_session_pool.py

```diff
@@ -0,0 +1,48 @@
+import pytest
+
+from ydb.query.pool import QuerySessionPool
+from ydb.query.session import QuerySessionSync, QuerySessionStateEnum
+
+
+class TestQuerySessionPool:
+    def test_checkout_provides_created_session(self, pool: QuerySessionPool):
+        with pool.checkout() as session:
+            assert session._state._state == QuerySessionStateEnum.CREATED
+
+        assert session._state._state == QuerySessionStateEnum.CLOSED
+
+    def test_oneshot_query_normal(self, pool: QuerySessionPool):
+        res = pool.execute_with_retries("select 1;")
+        assert len(res) == 1
+
+    def test_oneshot_ddl_query(self, pool: QuerySessionPool):
+        pool.execute_with_retries("create table Queen(key UInt64, PRIMARY KEY (key));")
+        pool.execute_with_retries("drop table Queen;")
+
+    def test_oneshot_query_raises(self, pool: QuerySessionPool):
+        with pytest.raises(Exception):
+            pool.execute_with_retries("Is this the real life? Is this just fantasy?")
```

Comment:
better use some neutral bad query

Reply:
better raise and check more specific error, for the example it will be ydb.BadRequest